eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala

```diff
@@ -383,10 +469,11 @@ private[channel] object ChannelCodecs3 {
 
   // Order matters!
   val stateDataCodec: Codec[HasCommitments] = discriminated[HasCommitments].by(uint16)
+    .typecase(0x09, Codecs.DATA_CLOSING_Codec)
     .typecase(0x08, Codecs.DATA_SHUTDOWN_Codec)
     .typecase(0x07, Codecs.DATA_NORMAL_Codec)
     .typecase(0x06, Codecs.DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT_Codec)
-    .typecase(0x05, Codecs.DATA_CLOSING_Codec)
+    .typecase(0x05, Codecs.DATA_CLOSING_COMPAT_05_Codec)
```

Comment:
I dropped the `option` codec tricks used in the previous commits, because I couldn't use it in the `revokedCommitPublished` case. This is much simpler.